src/app/widgets/NavbarConnection/index.jsx

```diff
@@ -366,6 +367,7 @@ class NavbarConnectionWidget extends PureComponent {
         const actions = {
             ...this.actions
         };
+        store.set('ports', ports);
```

Comment:
Why is this being stored in the store?  The store is for application preferences, not transient data.  Is there a reason you're not getting the list of port from controller.ports?